pkg/controller/lifecycle/actuator.go

```diff
@@ -66,8 +68,19 @@ func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, ex *extension
 		return err
 	}
 
+	pspDisabled := false
+	if a.serviceConfig.NetworkProblemDetector != nil {
+		if a.serviceConfig.NetworkProblemDetector.PSPDisabled != nil {
+			pspDisabled = *a.serviceConfig.NetworkProblemDetector.PSPDisabled
+		}
+	}
+
```

Comment:
The former behaviour still stands. I just moved it out.
But if it's cleaner your way, then let's go for it.